tex/sections/software.tex

```diff
@@ -81,18 +82,18 @@ \subsection{Direct methods}
     = 
     \, \cos(x^2) (2x).
 \end{equation}
-The stepsize dependence of the complex step differentiation method makes it resemble more to finite differences than AD with dual numbers. 
+The dependence of the complex step differentiation method on the step size gives it a closer resemblance to finite difference methods than to automatic differentiation (AD) using dual numbers.
 This difference between the methods also makes the complex step method sometimes more efficient than both finite differences and AD \cite{Lantoine_Russell_Dargent_2012}, an effect that can be counterbalanced by the number of extra unnecessary operations that complex arithmetic requires (see last column of \eqref{eq:AD-complex-comparision}) \cite{Martins_Sturdza_Alonso_2003_complex_differentiation}.
-It is also important to remark that many modern software already have support for complex number arithmetic, making complex step differentiation very easy to implement.
+It is also important to remark \todo{Repetition} that many modern software already have support for complex number arithmetic, making complex step differentiation very easy to implement.
 
 % \subsubsection{Reverse-mode AD}
 
-The libraries \texttt{ReverseDiff.jl} and \texttt{Zygote.jl} use callbacks to compute gradients. 
+The libraries \texttt{ReverseDiff.jl} and \texttt{Zygote.jl} use callbacks to compute gradients. \todo{We need to define what are callbacks}
 When gradients are being computed with less than $\sim 100$ parameters, the former is faster (see documentation).
 
 \cite{Baydin_Pearlmutter_Radul_Siskind_2015}
 
-Notice that the application of reverse AD on a numerical solver (without checkpointing) scales as $\mathcal O (n k)$, with $k$ the number of steps of the numerical solver. 
+Notice that the application of reverse AD on a numerical solver (without checkpointing) scales as $\mathcal O (n k)$, with $k$ the number of steps of the numerical solver. \todo{I think that we should introduce checkpointing before}
```

Comment:
I agree. It is still unclear to me where is the best part to introduce this.

Reply:
I opened an issue.